cucumber-expressions/ruby/spec/cucumber/cucumber_expressions/cucumber_expression_generator_spec.rb

```diff
@@ -162,6 +164,67 @@ class Currency
         expect(expressions[1].source).to eq("{zero-or-more} {zero-or-more} {zero-or-more}")
       end
 
+      context "defining 'force_full_word' on ParameterType" do
+        let(:force_full_word) { true }
+
+        let!(:direction_parameter_type) {
+          @parameter_type_registry.define_parameter_type(ParameterType.new(
+              'direction',
+              /(up|down)/,
+              String,
+              lambda {|s| s},
+              true,
+              false,
+              force_full_word
+          ))
+        }
+
+        context "when set to true, does not suggest parameter when match is" do
+          it "at the beginning of a word" do
+            expect(@generator.generate_expression("When I download a picture").source).not_to eq("When I {direction}load a picture")
+            expect(@generator.generate_expression("When I download a picture").source).to eq("When I download a picture")
+          end
+
+          it "inside a word" do
+            expect(@generator.generate_expression("When I watch the muppet show").source).not_to eq("When I watch the m{direction}pet show")
+            expect(@generator.generate_expression("When I watch the muppet show").source).to eq("When I watch the muppet show")
+          end
+
+          it "at the end of a word" do
+            expect(@generator.generate_expression("When I create a group").source).not_to eq("When I create a gro{direction}")
+            expect(@generator.generate_expression("When I create a group").source).to eq("When I create a group")
+          end
+
+          it "a full word" do
```

Comment:
this feels a separate context but not of big importance?

Reply:
I wanted to be sure not breaking anything there, but yes they are not that important checks indeed.

Maybe only keep the last one (which ensure that full words at the beginning and end of the sentence are matched)

Reply:
I think I may have been mis-interpreted.

The checks are fine. But I would wrap them as thus

```
context 'when set to true, does suggest parameter when a match is'
```

Reply:
ahh ok I got it now sorry. Yes I can do that :)